### ISSUE TITLE
disable-all button

### ISSUE THREAD

**Issue by jcb91** [Jupyter-contrib/jupyter_nbextensions_configurator#26]
in https://github.com/ipython-contrib/jupyter_contrib_nbextensions/issues/849, @ggrrll requested a button to disable all enabled nbextensions fromt he configurator page. This may prove misleading, since some nbextensions also come with serverextensions (e.g. the various conda environment ones), which wouldn't get disabled. But, if the button also shows a note to that effect, hopefully exessie confusion can be avoided...

**Comment by juhasch** [comment]
Now there is https://github.com/jupyter/jupyter_core/pull/96 which disables all extensions. I think an option just to disable nbextensions is a good ides, as you don't need to restart the notebook server.

What do you think about
* disabling all nbextensions vs. only contrib nbextensions. If we disable all, we need to allow turning them back on again.
* toggling nbextensions, i.e. storing the list of enabled extensions before disabling them, so they can be turned on again with one click.

**Comment by jcb91** [comment]
> disabling all nbextensions vs. only contrib nbextensions. If we disable all, we need to allow turning them back on again

Yeah, the problem with disabling contrib-only is that the configurator currently has no way to know which ones are which, so I'd just go with disabling all. Disabling as the configurator does it will leave all of them on the page (it sets the relevant `load_extensions` subkey to `false` ratehr than deleting it), so they can be restored ok.

> toggling nbextensions, i.e. storing the list of enabled extensions before disabling them, so they can be turned on again with one click.

Sounds potentially useful, but might require thoughtful handling, particularly if edits happen to the list in between the disable-all/restore-all - does the configurator restoree the exact state (disabling anything enabled in between) or just re-anable whatever was enabled when the store happened, ignoring anything enabled in the meantime? I think the latter seems more intuitive to me, but either way it'll need some explanation...
